Files.Launcher/DeviceWatcher.cs

```diff
@@ -31,6 +31,11 @@ public void Start()
             modifyWatcher = new ManagementEventWatcher(modifyQuery);
             modifyWatcher.EventArrived += new EventArrivedEventHandler(DeviceModifiedEvent);
             modifyWatcher.Start();
+            
+            WqlEventQuery cdQuery = new WqlEventQuery("SELECT * FROM __InstanceOperationEvent WITHIN 1 WHERE TargetInstance ISA 'Win32_CDROMDrive'");
```

Comment:
@gave92 I need some help here 😅 Will the above even work?

Reply:
@d2dyno1 Just took a look at this. AFAIK we're trying to avoid future use of the Microsoft.Management.Infrastructure library in the FullTrustProcess. The hope is to remove it all completely where we don't need it. 

With that said, there are some relatively robust WinRT APIs to send IOControlCodes to arbitrary devices. Let's leave this open for a bit while I experiment with them. We should theoretically be able to send a standard IOCTL_CDROM_LOAD_MEDIA code without the full trust process using [these](https://docs.microsoft.com/en-us/uwp/api/windows.devices.custom?view=winrt-22000) APIs and the optical drive capability.

Reply:
@d2dyno1 Actually, I was a little confused with what we need to use the FullTrustProcess to check in this PR. As long as we're not sending the IO control code to the device from the FullTrust component, something like this I wrote really quick: 

``` cs
private async Task<bool> CloseOpticalDiskTray()
        {
            Guid GUID_DEVINTERFACE_CDROM = new Guid("{53F56308-B6BF-11D0-94F2-00A0C91EFB8B}");
            string opticalDriveSelector = CustomDevice.GetDeviceSelector(GUID_DEVINTERFACE_CDROM);
            DeviceInformationCollection opticalDrives = await DeviceInformation.FindAllAsync(opticalDriveSelector, new string[] { "System.ItemPathDisplay" });
            foreach (DeviceInformation deviceInfo in opticalDrives)
            {
                string enumeratedPath = (string)deviceInfo.Properties["System.ItemPathDisplay"];
                string filename = @"\\.\" + driveLetter + ":";

                if (Path.GetPathRoot(enumeratedPath).Equals(Path.GetPathRoot(filename), StringComparison.OrdinalIgnoreCase))
                {
                    ushort FILE_DEVICE_CD_ROM = 0x00000002;

                    CustomDevice opticalDrive = await CustomDevice.FromIdAsync(deviceInfo.Id, DeviceAccessMode.ReadWrite, DeviceSharingMode.Shared);
                    IOControlCode IOCTL_CDROM_LOAD_MEDIA = new IOControlCode(FILE_DEVICE_CD_ROM, 0x0203, IOControlAccessMode.Read, IOControlBufferingMethod.Buffered);
                    return await opticalDrive.TrySendIOControlAsync(IOCTL_CDROM_LOAD_MEDIA, null, null);
                }
            }

            return false;
        }
```

while nicer to have, is likely unnecessary. If we have functionality in the FullTrustProcess that can't be replicated with WinRT APIs, then we can go ahead with this PR. 👍🏻

Reply:
This looks interesting, however, rewriting the entirety of `RemovableDevice` section would be out of the scope of this PR. That is not to say we shouldn't make a switch. If you'd like to you could open a PR with these changes, you seem more knowledgeable of these WinRT APIs than I am 😄

Reply:
@d2dyno1 Fair enough. I also understand your work better now, but I'll defer to @gave92's expertise on this one.

Reply:
I don't have a physical drive to test with (only daemontools virtual drive) but this should work.
I can think of two other ways to check is a drive is inserted from UWP:
1) Call `StorageHeldersExists("F:")` (F: being the cdrom drive letter)
2) Use DeviceIoControl with IOCTL_STORAGE_CHECK_VERIFY2: https://stackoverflow.com/questions/3158054/how-to-detect-if-media-is-inserted-into-a-removable-drive-card-reader

Reply:
The UWP alternative would be to use DeviceIoControl to check tray open/close status -> https://ws0.org/windows-how-to-get-the-tray-status-of-the-optical-drive/